src/plugins/data/common/search/aggs/agg_configs.ts

```diff
@@ -78,29 +79,27 @@ export type GenericBucket = estypes.AggregationsBuckets<any> & {
 export type IAggConfigs = AggConfigs;
 
 export class AggConfigs {
-  public indexPattern: IndexPattern;
   public timeRange?: TimeRange;
   public timeFields?: string[];
   public forceNow?: Date;
-  public hierarchical?: boolean = false;
-
-  private readonly typesRegistry: AggTypesRegistryStart;
-
-  aggs: IAggConfig[];
+  public aggs: IAggConfig[] = [];
+  public hierarchical?: boolean;
```

Comment:
nit: If we are touching this already, as `hierarchical` is always set in the constructor it can never be undefined

Reply:
@flash1293. I also thought that. But then reverted it back cause in `src/plugins/data/public/search/expressions/esaggs.ts `we have this kind of mutating: 

<img width="973" alt="image" src="https://user-images.githubusercontent.com/20072247/171853113-2e19a783-c66b-447c-a0c7-3055c859c3be.png">